protocols/luci-proto-wireguard/luasrc/model/cbi/admin_network/proto_wireguard.lua

```diff
@@ -81,10 +81,22 @@ preshared_key = section:taboption(
             "cryptography for post-quantum resistance.")
 )
 preshared_key.password = true
-preshared_key.datatype = "and(base64,rangelength(44, 44))"
+preshared_key.datatype = "and(base64,rangelength(44,44))"
 preshared_key.optional = true
 
 
+fwmark = section:taboption(
+  "advanced",
+  Value,
+  "fwmark",
+  translate("Firewall Mark"),
+  translate("Optional. 32-bit hexadecimal mark for outgoing encrypted " ..
+            "packets. Enter value including leading <code>0x</code>.")
+)
+fwmark.datatype = "and(hexstring,rangelength(0,8))"
```

Comment:
"Enter value in hex, starting with <code>0x</code>, or in decimal."

```
local datatypes  = require("luci.cbi.datatypes")
datatypes['hexordec'] = function(val)
	if val then
		if val:match("^0x[a-fA-F0-9]{1,8}$") ~= nil then
			return true
		end
		return datatypes["range"](val, 0, 4294967296)
	end
	return false
end

fwmark.datatype = "hexordec"
```
Dunno if this will work but it would be the most flexible situation.